File: java/src/sdk/base/paypal/payflow/ClientInfoHeader.java
package paypal.payflow;



/**
 * This class is used to store the complete information
 * about a client information header.
 */
final class ClientInfoHeader extends BaseRequestDataObject {
    /**
     * Stores Header name
     */
    private String headerName;

    /**
     * Stores Header value
     */
    private Object headerValue;

    /**
     * Contructor
     *
     * @param headerName  String
     * @param headerValue Object
     */
    public ClientInfoHeader(String headerName, Object headerValue) {
        this.headerName = headerName;
        this.headerValue = headerValue;
    }

    /**
     * Gets header name
     *
     * @return mHeaderName String
     */
    public String getHeaderName() {
        return this.headerName;
    }

    /**
     * Gets header value
     *
     * @return mHeaderValue Object
     */
    public Object getHeaderValue() {
        return this.headerValue;
    }
}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/ClientInfoHeader.cs
#region "Imports"
using System;
using PayPal.Payments.DataObjects;
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// This class is used to store the complete information 
	/// about a client information header.
	/// </summary>
	internal sealed class ClientInfoHeader : BaseRequestDataObject
	{
		/// <summary>
		/// Stores Header name
		/// </summary>
		private String mHeaderName;
		/// <summary>
		/// Stores Header value
		/// </summary>
		private Object mHeaderValue;
		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="HeaderName">Header name</param>
		/// <param name="HeaderValue">Header value</param>
		internal ClientInfoHeader(String HeaderName,Object HeaderValue)
		{
			mHeaderName = HeaderName;
			mHeaderValue = HeaderValue;
		}
		
		/// <summary>
		/// Gets header name
		/// </summary>
		internal String HeaderName
		{
			get { return mHeaderName; }
		}
		/// <summary>
		/// Gets header value
		/// </summary>
		internal Object HeaderValue
		{
			get { return mHeaderValue; }
		}
	}
}
